### ISSUE TITLE
"drupal init --override" fatal error

### ISSUE THREAD

**Issue by lukasfischer** [hechoendrupal/drupal-console#2745]
I installed DrupalConsole as noted on the website.

Drupal Console Launcher version 1.0.0-rc1

When I run drupal init, this php fatal error comes up:

```
drupal init --override

Warning: Invalid argument supplied for foreach() in phar:///usr/local/bin/drupal/vendor/drupal/console-core/src/ConsoleApplication.php on line 251

 Copy at current directory console/config.yml? (yes/no) [yes]:
 > 

 Enter drupal root directory [web, docroot] empty for none? [ ]:
 > 


Warning: array_keys() expects parameter 1 to be array, string given in phar:///usr/local/bin/drupal/vendor/drupal/console-core/src/Command/InitCommand.php on line 117

Catchable fatal error: Argument 2 passed to Drupal\Console\Style\DrupalStyle::choiceNoList() must be of the type array, null given, called in phar:///usr/local/bin/drupal/vendor/drupal/console-core/src/Command/InitCommand.php on line 118 and defined in phar:///usr/local/bin/drupal/vendor/drupal/console-core/src/Style/DrupalStyle.php on line 38
➜  ~ drupal --version

Warning: Invalid argument supplied for foreach() in phar:///usr/local/bin/drupal/vendor/drupal/console-core/src/ConsoleApplication.php on line 251
```

**Comment by msypes** [comment]
Additional information: I too, have received this error, but only after a second attempt to run `drupal init --override` in my home directory. 
On the first try, I was prompted for additional information: language, temp file path, learning info, inline representations, and chain representations.
The output included a list of copied files and series of PHP warnings, all about failing to open streams to twig files from ...Twig/Loader/Filesystem.php, line 131

**Comment by geerlingguy** [comment]
I seem to be getting this on a fresh Drupal VM install, too... log in after installing console 1.0.0-rc1, then if I run the init command and use defaults, it outputs:

```
Copied files
 User home path: /home/vagrant/.console/
 1 - aliases.yml
 2 - chain/create-data.yml
 3 - chain/form-sample.yml
 4 - chain/quick-start-db-env.yml
 5 - chain/quick-start-db.yml
 6 - chain/quick-start.yml
 7 - chain/sample.yml
 8 - chain/site-update.yml
 9 - chain/update-command-data.yml
 10 - chain/update-gitbook.yml
 11 - phpcheck.yml
 12 - router.php
 13 - site.mode.yml
 14 - sites/drupalvm.yml
 15 - sites/sample.yml

PHP Warning:  file_get_contents(phar:/usr/local/bin/drupal/vendor/drupal/console-core/templates/core/init/config.yml.twig): failed to open stream: No such file or directory in phar:///usr/local/bin/drupal/vendor/twig/twig/lib/Twig/Loader/Filesystem.php on line 131

Warning: file_get_contents(phar:/usr/local/bin/drupal/vendor/drupal/console-core/templates/core/init/config.yml.twig): failed to open stream: No such file or directory in phar:///usr/local/bin/drupal/vendor/twig/twig/lib/Twig/Loader/Filesystem.php on line 131
PHP Warning:  file_get_contents(phar:/usr/local/bin/drupal/vendor/drupal/console-core/templates/core/autocomplete/console.rc.twig): failed to open stream: No such file or directory in phar:///usr/local/bin/drupal/vendor/twig/twig/lib/Twig/Loader/Filesystem.php on line 131

Warning: file_get_contents(phar:/usr/local/bin/drupal/vendor/drupal/console-core/templates/core/autocomplete/console.rc.twig): failed to open stream: No such file or directory in phar:///usr/local/bin/drupal/vendor/twig/twig/lib/Twig/Loader/Filesystem.php on line 131
PHP Warning:  file_get_contents(phar:/usr/local/bin/drupal/vendor/drupal/console-core/templates/core/autocomplete/console.fish.twig): failed to open stream: No such file or directory in phar:///usr/local/bin/drupal/vendor/twig/twig/lib/Twig/Loader/Filesystem.php on line 131

Warning: file_get_contents(phar:/usr/local/bin/drupal/vendor/drupal/console-core/templates/core/autocomplete/console.fish.twig): failed to open stream: No such file or directory in phar:///usr/local/bin/drupal/vendor/twig/twig/lib/Twig/Loader/Filesystem.php on line 131
Bash or Zsh: Add this line to your shell configuration file:
source "$HOME/.console/console.rc" 2>/dev/null

Fish: Create a symbolic link
ln -s ~/.console/drupal.fish ~/.config/fish/completions/drupal.fish
```

Then if I run a simple command like `--version`, I get:

```
$ drupal --version
PHP Warning:  Invalid argument supplied for foreach() in phar:///usr/local/bin/drupal/vendor/drupal/console-core/src/ConsoleApplication.php on line 251

Warning: Invalid argument supplied for foreach() in phar:///usr/local/bin/drupal/vendor/drupal/console-core/src/ConsoleApplication.php on line 251
Drupal Console Launcher version 1.0.0-rc1
```

Reference to issue in Drupal VM: https://github.com/geerlingguy/drupal-vm/issues/909

Steps to reproduce:

  1. Download Drupal VM
  2. Uncomment `drupalconsole` in `installed_extras` in a custom `config.yml` file
  3. `vagrant up`
  4. Once machine is up, `vagrant ssh` in and run `drupal init`

**Comment by jurgenhaas** [comment]
Same error here. And if I skip the `--override` parameter, then it goes much further but fails with different errors later on:

```
PHP Warning:  file_get_contents(phar:/usr/local/bin/drupal/vendor/drupal/console-core/templates/core/init/config.yml.twig): failed to open stream: No such file or directory in phar:///usr/local/bin/drupal/vendor/twig/twig/lib/Twig/Loader/Filesystem.php on line 131
PHP Warning:  file_get_contents(phar:/usr/local/bin/drupal/vendor/drupal/console-core/templates/core/autocomplete/console.rc.twig): failed to open stream: No such file or directory in phar:///usr/local/bin/drupal/vendor/twig/twig/lib/Twig/Loader/Filesystem.php on line 131
PHP Warning:  file_get_contents(phar:/usr/local/bin/drupal/vendor/drupal/console-core/templates/core/autocomplete/console.fish.twig): failed to open stream: No such file or directory in phar:///usr/local/bin/drupal/vendor/twig/twig/lib/Twig/Loader/Filesystem.php on line 131
```

It creates an empty `~/console/config.yml` file and that'S also in the wrong directory, the leading period is missing.

**Comment by msypes** [comment]
The latest release (rc2) may have corrected this. I just did a self-update and re-ran `drupal init`. No errors this time.
I am still wary though as my home directory now has both a `.console` directory, and a `console` directory containing a config.yml. Also, this does not appear to have given me global access to the drupal console; i.e., calling it within a drupal root errors out that it's not installed there and that I should install it with composer.

**Comment by lukasfischer** [comment]
Yeah, rc2 fixed the issue.

As msypes mentioned, the config file is not copy to the right position. This is the output I got:
```
Generated or updated files
 Site path: phar:///usr/local/bin/drupal/bin/../
 1 - /Users/lukasfischer/console/config.yml
 2 - /Users/lukasfischer/.console/console.rc
 3 - /Users/lukasfischer/.console/drupal.fish
```

I move console/config.yml to .console/config.yml and all works fine.

**Comment by msypes** [comment]
I can confirm. Did the same thing. still can't use the console globally, which may be another issue.

**Comment by jmolivas** [comment]
@lukasfischer , @msypes , @geerlingguy , @jurgenhaas :

* This was related to a twig issue 
https://github.com/twigphp/Twig/issues/2145
https://github.com/twigphp/Twig/issues/2149

* Issue was fixed with this PR on the Twig project
https://github.com/twigphp/Twig/pull/2152

* A new Twig release was tagged
https://github.com/twigphp/Twig/releases/tag/v1.26.0

* A new DrupalConsole release was tagged including latest Twig version:
https://github.com/hechoendrupal/drupal-console-core/releases/tag/1.0.0-rc4
https://github.com/hechoendrupal/drupal-console-launcher/releases/tag/1.0.0-rc4
https://github.com/hechoendrupal/DrupalConsole/releases/tag/1.0.0-rc4

To execute the `init` command bypassing the interactive mode make sure you pass the `--no-interaction` option-flag
```
> drupal init --no-interaction

Bash or Zsh: Add this line to your shell configuration file:
source "$HOME/.console/console.rc" 2>/dev/null

Fish: Create a symbolic link
ln -s ~/.console/drupal.fish ~/.config/fish/completions/drupal.fish

Generated or updated files

 1 - /Users/jmolivas/.console/config.yml
 2 - /Users/jmolivas/.console/console.rc
 3 - /Users/jmolivas/.console/drupal.fish
```

**Comment by lukasfischer** [comment]
The config files seemed to be generated correctly. Howere if I run drupal comman in a D8 project root I get the message
```
[ERROR] Drupal Console is not installed at: /Users/lukasfischer/Sites/d8-site-test.com/
```

It seems that the drupal command cannot be executed globally or some config is missing. Any hint what I'm doing wrong here?

**Comment by jmolivas** [comment]
@lukasfischer is this a DrupalComposer based site? are you using DrupalVM?

If your site is based on DrupalComposer try executing `drupla init` and asweer yest top the very first question

`Copy at current directory console/config.yml? (yes/no) [yes]:`

**Comment by harold20** [comment]
This issue is not longer present in the current version 1.8.0
